pkg/webhook/webhook.go

```diff
@@ -138,34 +138,40 @@ func (wh *WebHook) serve(w http.ResponseWriter, r *http.Request) {
 	if r.Body != nil {
 		data, err := ioutil.ReadAll(r.Body)
 		if err != nil {
-			glog.Errorf("failed to read the request body")
-			http.Error(w, "failed to read the request body", http.StatusInternalServerError)
+			internalError(w, fmt.Errorf("failed to read the request body"))
 			return
 		}
 		body = data
 	}
 
 	if len(body) == 0 {
-		glog.Errorf("empty request body")
-		http.Error(w, "empty request body", http.StatusBadRequest)
+		denyRequest(w, "empty request body", http.StatusBadRequest)
 		return
 	}
 
 	contentType := r.Header.Get("Content-Type")
 	if contentType != "application/json" {
-		glog.Errorf("Content-Type=%s, expected application/json", contentType)
-		http.Error(w, "invalid Content-Type, expected `application/json`", http.StatusUnsupportedMediaType)
+		denyRequest(w, "invalid Content-Type, expected `application/json`", http.StatusUnsupportedMediaType)
 		return
 	}
 
 	var reviewResponse *admissionv1beta1.AdmissionResponse
 	review := &admissionv1beta1.AdmissionReview{}
 	deserializer := codecs.UniversalDeserializer()
 	if _, _, err := deserializer.Decode(body, nil, review); err != nil {
-		glog.Error(err)
-		reviewResponse = toAdmissionResponse(err)
+		internalError(w, err)
+		return
 	} else {
-		reviewResponse = mutatePods(review, wh.lister, wh.sparkJobNamespace)
+		if review.Request.Resource == podResource {
+			reviewResponse, err = mutatePods(review, wh.lister, wh.sparkJobNamespace)
+			if err != nil {
+				internalError(w, err)
+				return
+			}
+		} else {
+			denyRequest(w, fmt.Sprintf("Unexpected resource type: %v", review.Request.Resource.String()), http.StatusUnsupportedMediaType)
```

Comment:
We use lower-case for the first letter of error messages throughout the code base. Better to be consistent.